lib/web/integrations_awsoidc_test.go

```diff
@@ -421,3 +429,222 @@ func TestBuildListDatabasesConfigureIAMScript(t *testing.T) {
+	upsertDbSvcFn("vpc-2", nil)
+	upsertDbSvcFn("vpc-3", nil)
+	upsertDbSvcFn("vpc-4", nil)
+
+	// Test no required vpcs.
+	resp, err = awsOIDCRequiredVPCSHelper(ctx, req, mockListClient, clt)
+	require.NoError(t, err)
+	require.Empty(t, resp.VPCMapOfSubnets)
+}
+
+func TestAWSOIDCRequiredVPCSHelper_CombinedSubnetsForAVpcID(t *testing.T) {
+	ctx := context.Background()
+	env := newWebPack(t, 1)
+	clt := env.proxies[0].client
+
+	rdss := []rdsTypes.DBInstance{
+		{
+			DBInstanceStatus:     stringPointer("available"),
+			DBInstanceIdentifier: stringPointer("id-vpc1"),
+			DbiResourceId:        stringPointer("db-123"),
+			Engine:               stringPointer("postgres"),
+			DBInstanceArn:        stringPointer("arn:aws:iam::123456789012:role/MyARN"),
+
+			Endpoint: &rdsTypes.Endpoint{
+				Address: stringPointer("endpoint.amazonaws.com"),
+				Port:    aws.Int32(5432),
+			},
+			DBSubnetGroup: &rdsTypes.DBSubnetGroup{
+				Subnets: []rdsTypes.Subnet{
+					{SubnetIdentifier: aws.String("subnet1")},
+					{SubnetIdentifier: aws.String("subnet2")},
+				},
+				VpcId: aws.String("vpc-1"),
+			},
+		},
+		{
+			DBInstanceStatus:     stringPointer("available"),
+			DBInstanceIdentifier: stringPointer("id-vpc1a"),
+			DbiResourceId:        stringPointer("db-123"),
+			Engine:               stringPointer("postgres"),
+			DBInstanceArn:        stringPointer("arn:aws:iam::123456789012:role/MyARN"),
+
+			Endpoint: &rdsTypes.Endpoint{
+				Address: stringPointer("endpoint.amazonaws.com"),
+				Port:    aws.Int32(5432),
+			},
+			DBSubnetGroup: &rdsTypes.DBSubnetGroup{
+				Subnets: []rdsTypes.Subnet{
+					{SubnetIdentifier: aws.String("subnet2")},
+					{SubnetIdentifier: aws.String("subnet3")},
+					{SubnetIdentifier: aws.String("subnet4")},
+					{SubnetIdentifier: aws.String("subnet1")},
+				},
+				VpcId: aws.String("vpc-1"),
+			},
+		},
+		{
+			DBInstanceStatus:     stringPointer("available"),
+			DBInstanceIdentifier: stringPointer("id-vpc2"),
+			DbiResourceId:        stringPointer("db-123"),
+			Engine:               stringPointer("postgres"),
+			DBInstanceArn:        stringPointer("arn:aws:iam::123456789012:role/MyARN"),
+
+			Endpoint: &rdsTypes.Endpoint{
+				Address: stringPointer("endpoint.amazonaws.com"),
+				Port:    aws.Int32(5432),
+			},
+			DBSubnetGroup: &rdsTypes.DBSubnetGroup{
+				Subnets: []rdsTypes.Subnet{{SubnetIdentifier: aws.String("subnet8")}},
+
+				VpcId: aws.String("vpc-2"),
+			},
+		},
+	}
+
+	mockListClient := mockListDatabasesClient{dbInstances: rdss}
+
+	resp, err := awsOIDCRequiredVPCSHelper(ctx, ui.AWSOIDCRequiredVPCSRequest{Region: "us-east-1"}, mockListClient, clt)
+	require.NoError(t, err)
+	require.Len(t, resp.VPCMapOfSubnets, 2)
+	require.ElementsMatch(t, []string{"subnet1", "subnet2", "subnet3", "subnet4"}, resp.VPCMapOfSubnets["vpc-1"])
+	require.ElementsMatch(t, []string{"subnet8"}, resp.VPCMapOfSubnets["vpc-2"])
+}
+
+func stringPointer(s string) *string {
+	return &s
+}
```

Comment:
:nail_care: 
I think we should use `aws.String()` instead.
I used this method in the past, but given that it was only being used for AWS Types, I converted most of them to just use the auxiliary method from the AWS package